System/Vector.h

```diff
@@ -148,15 +148,15 @@ namespace RTE {
 		/// Indicates whether both X and Y components of this Vector are 0.
 		/// </summary>
 		/// <returns>Whether both X and Y components of this Vector are 0.</returns>
-		bool IsZero() const { return m_X == 0 && m_Y == 0; }
+		bool IsZero() const { return fabs(m_X) < m_FloatTolerance && fabs(m_Y) < m_FloatTolerance; }
 
 		/// <summary>
 		/// Indicates whether the X and Y components of this Vector each have opposite signs to their corresponding components of a passed in Vector.
 		/// </summary>
 		/// <param name="opp">The Vector to compare with.</param>
-		/// <returns>Whether both x and y components of this Vector are 0.</returns>
+		/// <returns>Whether the X and Y components of this Vector each have opposite signs to their corresponding components of a passed in Vector.</returns>
 		bool IsOpposedTo(const Vector &opp) { 
-			return (!m_X && !opp.m_X) || (m_X < 0 && opp.m_X > 0) || (m_X > 0 && opp.m_X < 0) && (!m_Y && !opp.m_Y) || (m_Y < 0 && opp.m_Y > 0) || (m_Y > 0 && opp.m_Y < 0); 
+			return ((fabs(m_X) < m_FloatTolerance && fabs(opp.m_X) < m_FloatTolerance) || (signbit(m_X) != signbit(opp.m_X))) && ((fabs(m_Y) < m_FloatTolerance && fabs(opp.m_Y) < m_FloatTolerance) || (signbit(m_Y) != signbit(opp.m_Y)));
```

Comment:
I wonder if it might be nice to have XIsZero() and YIsZero methods and use them here (and I guess in IsZero()) to make things more readable.
Not really worth it if we don't use it elsewhere though.

Reply:
in relation to what i wrote in the other comment, this can come into play.
maybe something like
```
ComponentIsZero(float comp){
fabs(comp) < m_FloatTolerance;
}
```
and then it could also be used for the == operator overloading, giving as input the difference between x components of two vectors.

Reply:
Yeah, you could do it that way if you like, I'm fine with that.

Reply:
That was an over-complication. i'm gonna do what you originally suggested.